e2e/deployers/retry.go

```diff
@@ -68,3 +76,88 @@ func WaitWorkloadHealth(ctx types.TestContext, cluster types.Cluster, namespace
 		}
 	}
 }
+
+func WaitForResourceDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	key k8stypes.NamespacedName,
+	obj client.Object,
+) error {
+	log := ctx.Logger()
+	resourceType := getResourceTypeName(obj)
+	startTime := time.Now()
+
+	log.Debugf("Waiting until %s \"%s/%s\" is deleted in cluster %q",
+		resourceType, key.Namespace, key.Name, cluster.Name)
```

Comment:
We want to support both namespace scope and cluster scope resources, so we should use a function to return the name for the log:

```go
func logName(obj client.Object) string {
    if obj.Namespace != "" {
        return obj.Namespace + "/" + obj.Name
    }
    return obj.Name
}